### PR TITLE
test: retry fqdn requests, increase curl timeout

### PR DESCRIPTION
**Pull request by nebril**

### CODE REVIEW

**Comment by nebril** [comment]
test-focus K8sFQDNTest

**Comment by nebril** [comment]
test-focus K8sFQDNTest

File: test/helpers/cons.go [hunk 1]
```diff
@@ -165,7 +165,7 @@ const (
 	// connecting or transferring data. CurlMaxTimeout should be at least 5
 	// seconds longer than CurlConnectTimeout to provide some time to
 	// actually transfer data.
-	CurlMaxTimeout = 8
+	CurlMaxTimeout = 20
```
Comment (qmonnet) [inline test/helpers/cons.go:168]: The timeout is used by `CurlFail()`, which is sometimes _supposed_ to fail. Won't that increase the waiting time for the tests when we expect the connexion to fail?

Comment (nebril) [reply test/helpers/cons.go:168]: @qmonnet this should be handled by `CurlConnectTimeout`, which is set to 5 (it maps to `--connect-timeout` flag), so this change shouldn't affect failing connection.

Comment (qmonnet) [reply test/helpers/cons.go:168]: Ah, my bad, all good then!

**Review by qmonnet — APPROVED** [review]

**Review by tklauser — COMMENTED** [review]
Should we also add `--retry 5` to the curl calls here

https://github.com/cilium/cilium/blob/0fa2bc77326eddea780e6257735829951fec4896/test/k8sT/fqdn.go#L249

and

https://github.com/cilium/cilium/blob/0fa2bc77326eddea780e6257735829951fec4896/test/k8sT/fqdn.go#L261

? (Sorry cannot comment inline, GH doesn't allow to add review comments to unchanged lines :cry: )

**Comment by nebril** [comment]
@tklauser these are supposed to fail, so it didn't flake on us (either there was a connectivity issue or Cilium worked as expected and blocked the request). I don't think there's need to add retries there.

**Comment by tklauser** [comment]
@nebril thanks for the explanation, makes sense.

**Review by tklauser — APPROVED** [review]

**Pull request merged**